domain/application/state/storage.go

```diff
@@ -630,6 +675,7 @@ WHERE  uuid = $storageUnit.storage_instance_uuid
 	// This only occurs in corner cases where a new pod appears with storage
 	// that needs to be reconciled with the Juju model. It is part of the
 	// UnitIntroduction workflow when a pod appears with volumes already attached.
+	// TODO - this can be removed when ObservedAttachedVolumeIDs are processed.
```

Comment:
Are we tracking this as part of that work in any way ?

Reply:
Yes, it's a JIRA card we've previously triaged together